### PR TITLE
Fix exit with bad state when backport branch already exists

### LINKED ISSUE #38: cherry-picker ends up in bad state if a `backport-...` branch already exists
If, for whatever reason, the `backport-...` branch already exists, cherry-picker ends up in a bad state:
```
λ python -m cherry_picker e5ee6a87a3b6fbf1a7e3856248885811f04d1f89 3.4
🐍 🍒 ⛏

Now backporting 'e5ee6a87a3b6fbf1a7e3856248885811f04d1f89' into '3.4'
Error checking out the branch backport-e5ee6a8-3.4.
fatal: A branch named 'backport-e5ee6a8-3.4' already exists.

λ python -m cherry_picker e5ee6a87a3b6fbf1a7e3856248885811f04d1f89 3.4
🐍 🍒 ⛏
Usage: __main__.py [OPTIONS] [COMMIT_SHA1] [BRANCHES]...
Try '__main__.py -h' for help.

Error: Run state cherry-picker.state=BACKPORT_LOOP_START in Git config is not known.
Perhaps it has been set by a newer version of cherry-picker. Try upgrading.
Valid states are: BACKPORT_PAUSED, UNSET. If this looks suspicious, raise an issue at https://github.com/python/core-workflow/issues/new.
As the last resort you can reset the runtime state stored in Git config using the following command: `git config --local --remove-section cherry-picker`
```

### PR DESCRIPTION
**Pull request by jack1142**
Fixes #38 
I also went ahead and added a `branch_name` attribute to `BranchCheckoutException` which should allow miss-islington (or others using the API) to remove the problematic branch automatically (see https://github.com/python/miss-islington/pull/480). I don't think it would be wise for cherry-picker to do it for you since that potentially risks you losing some work.

### CODE REVIEW

**Comment by ambv** [comment]
Your change is still worth integrating but note that when a backport branch already exists, now cherry-picker will force-push to it (with --force-with-lease so only if there are no other changes on that remote branch):

https://github.com/python/cherry-picker/commit/7122a15bec80326f3fbfa44a2dafa059749c9963

**Comment by jack1142** [comment]
Does that mean I should change this to overwrite the existing local branch or should I leave it as it is right now? I went for the more cautious way but I don't really mind changing it if you think it would be better the other way.

**Comment by jack1142** [comment]
I resolved the conflict caused by #56. Since I'm already commenting, I just want to gently remind you that I haven't received a response to my question here. I realize it's probably a terrible time for a reminder since it's Christmas, it just so happened that I found about these conflicts while working on something else today 😄 

Happy holidays!

File: cherry_picker/cherry_picker.py [hunk 1]
```diff
@@ -77,7 +77,9 @@
 
 
 class BranchCheckoutException(Exception):
-    pass
+    def __init__(self, branch_name):
+        self.branch_name = branch_name
+        super().__init__(f"Error checking out the branch {branch_name}.")
```
Comment (ezio-melotti) [inline cherry_picker/cherry_picker.py:82]:
```suggestion
        super().__init__(f"Error checking out the branch {branch_name!r}.")
```

File: cherry_picker/cherry_picker.py [hunk 2]
```diff
@@ -214,12 +216,10 @@ def checkout_branch(self, branch_name, *, create_branch=False):
             self.run_cmd(cmd)
         except subprocess.CalledProcessError as err:
             click.echo(
-                f"Error checking out the branch {branch_name}."
+                f"Error checking out the branch {checked_out_branch}."
```
Comment (ezio-melotti) [inline cherry_picker/cherry_picker.py:219]:
```suggestion
                f"Error checking out the branch {checked_out_branch!r}."
```

**Review by Mariatta — APPROVED** [review]

**Comment by Mariatta** [comment]
Thanks for the work.

**Pull request merged**